src/Config.h

```diff
@@ -0,0 +1,1488 @@
+         * @brief Load configuration from file
+         *
+         * @return true if successful, false otherwise
+         */
+        bool load() {
+            if (!LittleFS.exists(CONFIG_FILE)) {
+                LOG(INFO, "Config file does not exist");
+
+                return false;
+            }
+
+            File file = LittleFS.open(CONFIG_FILE, "r");
+
+            if (!file) {
+                LOG(ERROR, "Failed to open config file for reading");
+
+                return false;
+            }
+
+            const DeserializationError error = deserializeJson(_doc, file);
+            file.close();
+
+            if (error) {
+                LOG(ERROR, "Failed to parse config file");
+                return false;
+            }
+
+            LOG(INFO, "Configuration loaded successfully");
+
+            return true;
+        }
+
+        /**
+         * @brief Save configuration to file
+         *
+         * @return true if successful, false otherwise
+         */
+        [[nodiscard]] bool save() const {
+            File file = LittleFS.open(CONFIG_FILE, "w");
+
+            if (!file) {
+                LOG(ERROR, "Failed to open config file for writing");
+                return false;
+            }
+
+            if (serializeJson(_doc, file) == 0) {
+                LOG(ERROR, "Failed to write config to file");
+                file.close();
+                return false;
+            }
+
+            file.close();
+            LOG(INFO, "Configuration saved successfully");
+
+            return true;
+        }
+
+        /**
+         * @brief Reset configuration to factory defaults
+         *
+         * @return true if successful, false otherwise
+         */
+        bool factoryReset() {
+            createDefaults();
+            return save();
+        }
+
+        bool validateAndApplyFromJson(const String& jsonString) {
+            DynamicJsonDocument doc(4096);
+            const DeserializationError error = deserializeJson(doc, jsonString);
+
+            if (error) {
+                LOGF(ERROR, "JSON parsing failed: %s", error.c_str());
+                return false;
+            }
+
+            if (!validateAndApplyConfig(doc)) {
+                return false;
+            }
+
+            return true;
+        }
+
+        // WiFi credentials flag
+        bool getWifiCredentialsSaved() {
+            return _doc["wifi"]["credentials_saved"] | false;
+        }
+
+        void setWifiCredentialsSaved(const bool value) {
+            _doc["wifi"]["credentials_saved"] = value;
+        }
+
+        // PID general
+        bool getPidEnabled() {
+            return _doc["pid"]["enabled"] | false;
+        }
+
+        void setPidEnabled(const bool value) {
+            _doc["pid"]["enabled"] = value;
+        }
+
+        bool getUsePonM() {
+            return _doc["pid"]["use_ponm"] | false;
+        }
+
+        void setUsePonM(const bool value) {
+            _doc["pid"]["use_ponm"] = value;
```

Comment:
An even cleaner solution than this would be to have a `PidConfig` class which has `getUsePonM()`, and `Config::getPidConfig()`